src/main/java/de/diddiz/util/MySQLConnectionPool.java

```diff
@@ -1,437 +1,39 @@
-import java.sql.CallableStatement;
-import java.sql.Clob;
 import java.sql.Connection;
-import java.sql.DatabaseMetaData;
-import java.sql.DriverManager;
-import java.sql.NClob;
-import java.sql.PreparedStatement;
-import java.sql.SQLClientInfoException;
 import java.sql.SQLException;
-import java.sql.SQLWarning;
-import java.sql.SQLXML;
-import java.sql.Savepoint;
-import java.sql.Statement;
-import java.sql.Struct;
-import java.util.Enumeration;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Properties;
-import java.util.Vector;
-import java.util.concurrent.Executor;
-import java.util.concurrent.locks.Lock;
-import java.util.concurrent.locks.ReentrantLock;
 
 public class MySQLConnectionPool implements Closeable
 {
-	private final static int poolSize = 10;
-	private final static long timeToLive = 300000;
-	private final Vector<JDCConnection> connections;
-	private final String url, user, password;
-	private final Lock lock = new ReentrantLock();
+
+	private final HikariDataSource ds;
 
 	public MySQLConnectionPool(String url, String user, String password) throws ClassNotFoundException {
-		Class.forName("com.mysql.jdbc.Driver");
-		this.url = url;
-		this.user = user;
-		this.password = password;
-		connections = new Vector<JDCConnection>(poolSize);
-        ConnectionReaper reaper = new ConnectionReaper();
-		new Thread(reaper, "MySQL Connection Reaper Thread - LogBlock").start();
+		this.ds = new HikariDataSource();
+		ds.setJdbcUrl(url);
+		ds.setUsername(user);
+		ds.setPassword(password);
+
+		ds.addDataSourceProperty("useUnicode", "true");
+		ds.addDataSourceProperty("characterEncoding", "utf-8");
+		ds.addDataSourceProperty("rewriteBatchedStatements", "true");
+
+		ds.addDataSourceProperty("cachePrepStmts", "true");
+		ds.addDataSourceProperty("prepStmtCacheSize", "250");
+		ds.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		ds.addDataSourceProperty("useServerPrepStmts", "true");
 	}
 
 	@Override
 	public void close() {
-		lock.lock();
-		final Enumeration<JDCConnection> conns = connections.elements();
-		while (conns.hasMoreElements()) {
-			final JDCConnection conn = conns.nextElement();
-			connections.remove(conn);
-			conn.terminate();
-		}
-		lock.unlock();
+		ds.close();
 	}
 
 	public Connection getConnection() throws SQLException {
-		lock.lock();
-		try {
-			final Enumeration<JDCConnection> conns = connections.elements();
-			while (conns.hasMoreElements()) {
-				final JDCConnection conn = conns.nextElement();
-				if (conn.lease()) {
-					if (conn.isValid())
-						return conn;
-					connections.remove(conn);
-					conn.terminate();
-				}
-			}
-			final JDCConnection conn = new JDCConnection(DriverManager.getConnection(url, user, password));
-			conn.lease();
-			if (!conn.isValid()) {
-				conn.terminate();
-				throw new SQLException("Failed to validate a brand new connection");
-			}
-			connections.add(conn);
-			if (mb4) conn.createStatement().executeQuery("SET NAMES utf8mb4");

```

Comment:
See lines 94 and 95 of [LogBlock.java](https://github.com/LogBlock/LogBlock/blob/master/src/main/java/de/diddiz/LogBlock/LogBlock.java#L94-L95)

This is from commit 82b4ffc2a28eef2d45713bbf5726d7c31d44e2d4 which adds support for the full set of Unicode points.  I found I needed to explicitly do that query or, even if the database column supported it, it still wouldn't allow them

Not sure we'd set that